cmux.go

```diff
@@ -177,7 +177,8 @@ func (l muxListener) Accept() (net.Conn, error) {
 // MuxConn wraps a net.Conn and provides transparent sniffing of connection data.
 type MuxConn struct {
 	net.Conn
-	buf buffer
+	buf     bytes.Buffer
```

Comment:
well, actually I think it's doable :-) https://github.com/soheilhy/cmux/commit/04d6a17e41

it also simplifies the shenanigan I had in `MuxConn.Read()`. WDYT?